### DISCUSSION TITLE
ICS-20 fungible token transfer denom metadata

### DISCUSSION THREAD

**Discussion by michaelfig** [cosmos/ibc#607] [Ideas]
It is a general problem that user interfaces for a chain need to at least understand some metadata (number of decimal places, preferred name, symbol, etc).  These concerns are addressed on Cosmos by the [coin metadata ADR](https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-024-coin-metadata.md), and on Agoric by our [ERTP brand displayInfo](https://agoric.com/documentation/ertp/api/brand.html#brand-getdisplayinfo).

I think it would be desirable for ICS-20 to be extended by a subprotocol that can convey this information over the transfer protocol.  Attaching the metadata to transferred tokens is a sane default, just like ICS-20 already uses a single piece of metadata: the denomination.  The issue of trusting the metadata is separate, but it is easier to say "attach the conveyed metadata to the token, and decide later if you value the token in the same way" rather than "default to no metadata, and require governance or a registry to install an ex nihilo idea of what the metadata should be".

I think chain token metadata interop would greatly improve the ergonomics for new tokens transferred to a given chain.  Because we lack this interop, the Agoric chain essentially needs a governance action for the first of what should be a permissionless token transfer.  This is because Agoric's token metadata is deeply wired into the chain, and every asset transfer conveys it.  In fact, the number of decimal places is considered even more fundamental in Agoric than the denomination name; we use "petnames" for asset types assigned by the user rather than a centrally-controlled name.

I hope to discuss this in a future IBC meeting, @AdityaSripal.  Also tagging @JimLarson, @dtribble, @warner on the Agoric side.

**Comment by hxrts** [comment]
Also relevant to the following efforts:

- https://github.com/osmosis-labs/assetlists
- https://github.com/cosmos/registry + https://github.com/cosmos/registry/issues/25
- And eventually https://github.com/tendermint/cns

**Comment by michaelfig** [comment]
Moving discussion to #719.
